pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -182,11 +182,7 @@ func Pod(source *operatorsv1alpha1.CatalogSource, name string, image string, saN
 						},
 					},
 					SecurityContext: &corev1.SecurityContext{
-						ReadOnlyRootFilesystem:   pointer.Bool(false),
-						AllowPrivilegeEscalation: pointer.Bool(false),
```

Comment:
Also, because we need to read root file system, they away to grant permissions to that is setting true to AllowPrivilegeEscalation

And RunAsUser=0 is not required ( we need an user that have permissions to do what needs to be done as before, and if the bt setting true there the user will be part of the root group and have the same permissions.

Btw, RunAsUser=0 might not work on OCP (it will depends of the namespace and it probably will raise the error invalid value in the range)

c/c @perdasilva

Reply:
what does "AllowPrivilegeEscalation" have to do w/ reading the root filesystem?

allow privilege escalation means you can run a setuid binary or otherwise allow a process to "become root", does the opm registry binary do so?  

I would simply expect the index.db to be readable (by any uid), or in general we should be ensuring that the container is run w/ the appropriate uid and fsgroup to allow us to read (and write, if necessary) to the appropriate directories.  None of that should require privilege escalation.


```
allowPrivilegeEscalation: Controls whether a process can gain more privileges than its parent process.
```

https://kubernetes.io/docs/tasks/configure-pod-container/security-context/

Reply:
That'd mean we don't need to specify `AllowPrivilegeEscalation` at all then, since opm doesn't require to be root to read or write to the sqllite db, and also in 4.10 we didn't specify `AllowPrivilegeEscalation` either (and opm not requiring to be root hasn't changed since 4.10, and is unlikely to change either).

Reply:
Fyi was updating the tests to use image hosted in the quay.io/olmtest registry instead of my personal registry, so took that opportunity to remove `AllowPrivilegeEscalation:true` too. cc: @perdasilva now I'm happy with the PR going in :)